crypto/ec/ecp_mont.c

```diff
@@ -206,6 +207,54 @@ int ec_GFp_mont_field_sqr(const EC_GROUP *group, BIGNUM *r, const BIGNUM *a,
     return BN_mod_mul_montgomery(r, a, a, group->field_data1, ctx);
 }
 
+/*-
+ * Computes the multiplicative inverse of a in GF(p), storing the result in r.
+ * If a is zero (or equivalent), you'll get a EC_R_CANNOT_INVERT error.
+ * We have a Mont structure, so SCA hardening is FLT inversion.
+ */
+int ec_GFp_mont_field_inv(const EC_GROUP *group, BIGNUM *r, const BIGNUM *a,
+                            BN_CTX *ctx)
+{
+    BIGNUM *e = NULL;
+    BN_CTX *new_ctx = NULL;
+    int ret = 0;
+
+    if (group->field_data1 == NULL)
+        return 0;
+
+    if (ctx == NULL && (ctx = new_ctx = BN_CTX_secure_new()) == NULL)
+        return 0;
+
+    BN_CTX_start(ctx);
+    if ((e = BN_CTX_get(ctx)) == NULL)
+        goto err;
+
+    /* Inverse in constant time with Fermats Little Theorem */
+    if (!BN_set_word(e, 2))
+        goto err;
+    if (!BN_sub(e, group->field, e))
+        goto err;
+    /*-
+     * Exponent e is public.
+     * No need for scatter-gather or BN_FLG_CONSTTIME.
```

Comment:
Not sure I understand the scatter-gather comment. Can you explain?

Reply:
It's @bbbrumley comment, but I would say that he meant that when inverting through FLT, we do `a^(p-2)`: the exponentiation could leak the exponent, which is public anyway, so there is no need to use `BN_FLG_CONSTTIME` to do a const-time exponentiation nor to somehow blind the scalar `a`.

Reply:
@mattcaswell Read the comments in #6116

> To that end, [e913d11](https://github.com/openssl/openssl/commit/e913d11f444e0b46ec1ebbf3340813693f4d869d) was not really correct

Reply:
What's more, to have a chuckle at #8253. Performance wise, no one will notice if the flag is set or not. At some point #6116 was randomly inheriting the flag anyway. I'm not sure about the current state.

Reply:
> @mattcaswell Read the comments in #6116
> 
> > To that end, [e913d11](https://github.com/openssl/openssl/commit/e913d11f444e0b46ec1ebbf3340813693f4d869d) was not really correct

I had a hard time finding the thread in the comments of #6116. This should be the conversation @bbbrumley is pointing at: https://github.com/openssl/openssl/pull/6116#discussion_r186595991

Reply:
Scatter-gather is probably not the best descriptor of `BN_mod_exp_mont` vs `BN_mod_exp_mont_consttime`. It's whether you use fixed windows with secret window values, or variable windows with public window values. Scatter-gather is a particular implementation strategy for the table lookup given secret window values. (And, depending on how it's done, [can be questionable](https://eprint.iacr.org/2016/224.pdf). Best to exhibit the same memory access pattern even within a cacheline.)